internal/jwk/key_test.go

```diff
@@ -0,0 +1,177 @@
+package jwk_test
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/acronis/go-authkit/internal/jwk"
+)
+
+func encodeBigIntToBase64URL(n *big.Int) string {
+	return base64.RawURLEncoding.EncodeToString(n.Bytes())
+}
+
+func TestDecodePublicKey(t *testing.T) {
+	// Define meaningful primes
+	p := big.NewInt(61)
+	q := big.NewInt(53)
+
+	// Compute modulus n = p * q
+	n := new(big.Int).Mul(p, q) // n = 61 × 53 = 3233
+	e := big.NewInt(65537)      // common public exponent
+
+	key := &jwk.Key{
+		Kty: "RSA",
+		N:   encodeBigIntToBase64URL(n),
+		E:   encodeBigIntToBase64URL(e),
+	}
+
+	pubKey, err := key.DecodePublicKey()
+	assert.NoError(t, err)
```

Comment:
It would be better to use the `require` package instead of `assert` here and in other places where the test should stop execution and fail immediately. Otherwise, if an error occurs, execution will continue, potentially leading to a panic in later lines, making it harder to understand the root cause of the test failure.